dev-tools/omdb/src/bin/omdb/mgs/sensors.rs

```diff
@@ -0,0 +1,954 @@
+
+    timestamps.push(std::time::Instant::now());
+
+    //
+    // First, get a component list.
+    //
+    let components = mgs_client.sp_component_list(type_, slot).await?;
+    timestamps.push(std::time::Instant::now());
+
+    //
+    // Now, for every component, we're going to get its details:  for those
+    // that are sensors (and contain measurements), we will store the name
+    // of the sensor as well as the retrieved value.
+    //
+    for c in &components.components {
+        for s in mgs_client
+            .sp_component_get(type_, slot, &c.component)
+            .await?
+            .iter()
+            .filter_map(|detail| match detail {
+                SpComponentDetails::Measurement { kind, name, value } => Some(
+                    (Sensor { name: name.clone(), kind: *kind }, Some(*value)),
+                ),
+                SpComponentDetails::MeasurementError { kind, name, error } => {
+                    match error {
+                        MeasurementErrorCode::NoReading
+                        | MeasurementErrorCode::NotPresent => None,
+                        _ => Some((
+                            Sensor { name: name.clone(), kind: *kind },
+                            None,
+                        )),
+                    }
+                }
+                _ => None,
+            })
+        {
+            devices.insert(DeviceIdentifier::Device(c.component.clone()), s);
+        }
+    }
+
+    timestamps.push(std::time::Instant::now());
+
+    Ok(SpInfo { devices, timestamps })
+}
+
+async fn sp_info_mgs(
+    mgs_client: &gateway_client::Client,
+    args: &SensorsArgs,
+) -> Result<SensorSpInfo, anyhow::Error> {
+    let mut rval = vec![];
+    let mut latencies = HashMap::new();
+
+    //
+    // First, get all of the SPs that we can see via Ignition
+    //
+    let all_sp_list =
+        mgs_client.ignition_list().await.context("listing ignition")?;
+
+    let mut sp_list = all_sp_list
+        .iter()
+        .filter_map(|ignition| {
+            if matches!(ignition.details, SpIgnition::Yes { .. }) {
+                if ignition.id.type_ == SpType::Sled {
+                    if args.matches_sp(&ignition.id) {
+                        return Some(ignition.id);
+                    }
+                }
+            }
+            None
+        })
+        .collect::<Vec<_>>();
+
+    if args.switches {
+        sp_list.push(SpIdentifier { type_: SpType::Switch, slot: 0 });
+        sp_list.push(SpIdentifier { type_: SpType::Switch, slot: 1 });
+    }
+
+    if args.psc {
+        sp_list.push(SpIdentifier { type_: SpType::Power, slot: 0 });
+    }
+
+    sp_list.sort();
+
+    let now = std::time::Instant::now();
+
+    let mut handles = vec![];
+    for sp_id in sp_list {
+        let mgs_client = mgs_client.clone();
+        let type_ = sp_id.type_;
+        let slot = sp_id.slot;
+
+        let handle =
+            tokio::spawn(async move { sp_info(mgs_client, type_, slot).await });
```

Comment:
Tiny nit: `sp_info` returns a future that can be spawned directly; we don't have to wrap it in an async block:

```suggestion
            tokio::spawn(sp_info(mgs_client, type_, slot));
```

(I think after making this change you could probably drop the 3 locals too, and make it `spawn(sp_info(mgs_client.clone(), sp_id.type_, sp_id.slot))`, since you no longer need to move them into a block.)